homeassistant/components/tado/climate.py

```diff
@@ -151,47 +129,61 @@ class TadoClimate(ClimateDevice):
 
     def __init__(
         self,
-        store,
+        tado,
         zone_name,
         zone_id,
-        data_id,
+        zone_type,
         min_temp,
         max_temp,
         step,
-        ac_device,
-        hot_water_device,
         ac_support_heat,
-        tolerance=0.3,
+        fallback,
     ):
         """Initialize of Tado climate device."""
-        self._store = store
-        self._data_id = data_id
+        self._tado = tado
 
         self.zone_name = zone_name
         self.zone_id = zone_id
+        self.zone_type = zone_type
 
-        self._ac_device = ac_device
-        self._hot_water_device = hot_water_device
+        self._ac_device = zone_type == TYPE_AIR_CONDITIONING
         self._ac_support_heat = ac_support_heat
         self._cooling = False
 
         self._active = False
         self._device_is_active = False
 
-        self._unit = TEMP_CELSIUS
         self._cur_temp = None
         self._cur_humidity = None
         self._is_away = False
         self._min_temp = min_temp
         self._max_temp = max_temp
         self._step = step
         self._target_temp = None
-        self._tolerance = tolerance
+
+        if fallback:
+            _LOGGER.debug("Default overlay is configured in TADO MODE")
+            # Fallback to Smart Schedule at next Schedule switch
+            self._default_overlay = CONST_OVERLAY_TADO_MODE
+        else:
+            _LOGGER.debug("Default overlay is configured in MANUAL MODE")
+            # Don't fallback to Smart Schedule, but keep in manual mode
+            self._default_overlay = CONST_OVERLAY_MANUAL
 
         self._current_fan = CONST_MODE_OFF
         self._current_operation = CONST_MODE_SMART_SCHEDULE
         self._overlay_mode = CONST_MODE_SMART_SCHEDULE
 
+    async def async_added_to_hass(self):
+        """Register for sensor updates."""
+        async_dispatcher_connect(
+            self.hass,
+            SIGNAL_TADO_UPDATE_RECEIVED.format("zone", self.zone_id),
+            self._handle_update,
+        )
+        # Process first update
+        await self.hass.async_add_executor_job(self._handle_update)
```

Comment:
If we rename `_handle_update` to `update` we can remove this call here.

Reply:
Does update get called with should_poll=False?

Reply:
Not after entity addition, but since we call `add_entities` with second argument `True`, `update` will be called once during entity addition.

Reply:
Hmm, I've renamed it to `update`, and now the update method indeed gets called after creation due to `add_entities`, but something is going wrong there:

```
2019-12-20 10:54:06 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/home/michael/Development/home-assistant/homeassistant/helpers/entity.py", line 273, in async_update_ha_state
    f"No entity id specified for entity {self.name}"
homeassistant.exceptions.NoEntitySpecifiedError: No entity id specified for entity Heating
```

It seems that the entity doesn't have an id yet, but when `async_added_to_hass` is called, it has one. Is it possible that hass is calling the `update` (due to the boolean in `add_entities`) method to soon?

Reply:
This seems related to this: https://github.com/home-assistant/home-assistant/pull/12277/files/e50ee974c0d2f84890aa0ba1853cc834b4fcd98d#r169013308